Sources/Controllers/MyPlaces/TracksViewController.swift

```diff
@@ -221,119 +242,158 @@ class TracksViewController: OACompoundViewController, UITableViewDelegate, UITab
-            } else {
-                if savingHelper.hasData() {
-                    let currentPausedTrackRow = section.createNewRow()
-                    currentPausedTrackRow.cellType = OATwoButtonsTableViewCell.reuseIdentifier
-                    currentPausedTrackRow.key = recordingTrackKey
-                    currentPausedTrackRow.title = localizedString("recorded_track")
-                    currentPausedTrackRow.descr = getTrackDescription(distance: savingHelper.distance, timeSpan: savingHelper.getCurrentGPX().timeSpan, waypoints: savingHelper.points, showDate: false, filepath: nil)
-                    let isVisible = settings.mapSettingShowRecordingTrack.get()
-                    currentPausedTrackRow.setObj(isVisible, forKey: isVisibleKey)
-                    currentPausedTrackRow.iconName = "ic_custom_track_recordable"
-                    currentPausedTrackRow.iconTintColor = isVisible ? UIColor.iconColorActive : UIColor.iconColorDefault
-                    currentPausedTrackRow.setObj(localizedString("ic_custom_play"), forKey: buttonIconKey)
-                    currentPausedTrackRow.setObj(ButtonActionNumberTag.startRecording.rawValue, forKey: buttonActionNumberTagKey)
-                    currentPausedTrackRow.setObj(localizedString("ic_custom_download"), forKey: secondButtonIconKey)
-                    currentPausedTrackRow.setObj(ButtonActionNumberTag.save.rawValue, forKey: secondButtonActionNumberTagKey)
-                } else {
-                    let recordNewTrackRow = section.createNewRow()
-                    recordNewTrackRow.cellType = OAButtonTableViewCell.reuseIdentifier
-                    recordNewTrackRow.key = recordingTrackKey
-                    recordNewTrackRow.title = localizedString("new_track")
-                    recordNewTrackRow.descr = localizedString("not_recorded")
-                    recordNewTrackRow.iconName = "ic_custom_trip"
-                    recordNewTrackRow.iconTintColor = UIColor.iconColorDefault
-                    recordNewTrackRow.setObj(localizedString("start_recording"), forKey: buttonTitleKey)
-                    recordNewTrackRow.setObj(localizedString("ic_custom_play"), forKey: buttonIconKey)
-                    recordNewTrackRow.setObj(ButtonActionNumberTag.startRecording.rawValue, forKey: buttonActionNumberTagKey)
-                    let isVisible = settings.mapSettingShowRecordingTrack.get()
-                    recordNewTrackRow.setObj(isVisible, forKey: isVisibleKey)
-                    recordNewTrackRow.setObj(isVisible ? UIColor.iconColorActive : UIColor.iconColorDefault, forKey: colorKey)
+        if isFiltered {
+            var foundedFolders: [TrackFolder] = []
+            var foundedTracks: [OAGPX] = []
+            for folder in rootFolder.flattenedFolders.values {
+                if folder.path.lastPathComponent().lowercased().contains(searchText.lowercased()) {
```

Comment:
private func containsCaseInsensitive(text: String, substring: String) -> Bool {
        text.range(of: substring, options: .caseInsensitive) != nil
    }
and create string extension